canvas_modules/common-canvas/src/object-model/api-pipeline.js

```diff
@@ -936,6 +958,73 @@ export default class APIPipeline {
 		return movedNodesInfo;
 	}
 
+	// Returns an array of move link actions that can be used to reposition the
+	// links based on the provided Dagre output graph.
+	// Iterate nodes and check if there are any dummy nodes.
+	// If there are dummy nodes then links should have x/y cordinates as per dummy_nodes.
+	convertGraphToMoveLinks(outputGraph, canvasInfoPipelineLinks) {
+		const movedLinksInfo = {};
+		const linksPos = [];
+
+		const nodes = outputGraph.nodes;
+
+		nodes.forEach((node) => {
+			// Record the x and y position of dummy node to be assigned to the detached link.
+			// A fully detached link can have both temp-src-node and temp-trg-node
+			if (node.v.startsWith("temp-src-node-")) {
+				const dummyNodeId = node.v.split("temp-src-node-")[1];
+
+				linksPos[dummyNodeId] = {
+					...linksPos[node.v.split("temp-src-node-")[1]],
+					srcPos: {
+						x: node.value.x,
+						y: node.value.y
+					}
+				};
```

Comment:
This seems a little over-complicated. It seems like you are getting the linkId again 
```
node.v.split("temp-src-node-")[1]
```
which you already got in the line above -- and then referencing into an element of the link array that will not have anything in it. (Although in the next section down for target Pos it may have something in it if we are handling a fully-detached link).